src/librustc_resolve/lib.rs

```diff
@@ -4196,17 +4196,15 @@ impl<'a, 'crateloader: 'a> Resolver<'a, 'crateloader> {
                 if ident.name == lookup_name && ns == namespace {
                     if filter_fn(name_binding.def()) {
                         // create the path
-                        let mut segms = if self.session.rust_2018() && !in_module_is_extern {
+                        let mut segms = path_segments.clone();
+                        if self.session.rust_2018() && !in_module_is_extern {
```

Comment:
btw I think we should remove the `!in_module_is_extern` flag here -- this way, if somehow there *is* an extern crate `foo` that is not part of the prelude -- then we will print `crate::foo::bar`, which I think is correct, since just `foo::bar` would not work

Reply:
Shouldn't that be `::foo::bar` instead, *without* `extern crate`? At least for now it works regardless of whether `foo` is in the `extern_prelude`.